### ISSUE TITLE
Glean logs do not show up on VPN exported logs

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#4643]
**Mozilla VPN affected versions:**

* 2.10.0 (16655930);

**Platform tested:**

* Android 10;

**Prerequisites:**

* Log in with an account with subscription for Mozilla VPN;
* Open the logs;

**Steps to reproduce:**

1. Search for a Glean PING;

**Expected results:**

* A ping ID is found. Using it a Glean event can be seen in BigQuery database;

**Actual results:**

* There is no ping ID in the logs.

**Notes:**

* Not reproducing on the other platforms;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-3019)
┆Reporter: Valentina Virlics
┆Link To Issue: https://mozilla-hub.atlassian.net/browse/VPN-3019

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Since Android is not using the Glean JavaScript SDK but the Glean Kotlin SDK this might just be an issue of Glean logs not showing up in the VPN logs for Android. I noticed a similar issue when experimenting with the Glean Rust SDK https://mozilla-hub.atlassian.net/browse/VPN-2724 ( https://mozilla-hub.atlassian.net/browse/VPN-2724|smart-link ), where Glean logs are ommited from the VPN logs.

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

[^out.txt]

I was able to check that indeed the problem is in the VPN logs. Please see attached logs on line 389 a ping is sent.

Valentina Virlics if you have [adb|] installed you can see these same logs using adb logcat | grep -e 'mozillavpn' -e 'glean'.

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

Based on the latest comments, this works as expected. Closing the bug.

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

This actually happens for all platforms, it’s an issue with our logs and the Glean Rust logs.

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Hey Valentina Virlics ! So… This is merged on main. I would like to get it uplifted to the release branch, but it’s not there yet.

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

Since this was merged in January, I’m guessing it’s only in the Dev branch (2.14), right Beatriz Rizental Machado ?

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

I have requested this be uplifted to 2.13. Changing the fix versions. Will mark it as”needs QA” once it is merged. Let me know if that process is incorrect.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Beatriz Rizental Machado So, this can be verified using latest 2.13? Thank you!

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Valentina Virlics Correct. Just merged.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Beatriz Rizental Machado After a fix is uplifted or directly merged into release branch, we move the ticket in Merged state and add the 2.13 (current release) fixVersion.

Verified this issue, while using the 2.13.0 (2.202301121719) build on Android 10, the Glean pigs are recorded in the client Logs, not sure if they are correct, because we tried different pings and for none of the we have events in mozdata - Big Query ( https://console.cloud.google.com/bigquery?sq=470293616514:1e35cd4514a04d1ca1c3c40ee61f8221&authuser=1&pli=1&rapt=AEjHL4OUyHZAzRrzDOHYIdm4n2oT231Z88C62QIW1S8ujdc46Zii24PiaktqmoqIV6MLzWOi0pzJjjIMbOpHqElOC8has8Wn6g&project=mozdata&supportedpurview=project&ws=!1m5!1m4!1m3!1smozdata!2sbquxjob_45428b17_185aab0dcfa!3sUS ) - project.

Used query below:

!image-20230113-103848.png|width=883,height=699!

Selected ping from the Logs:

!image-20230113-103825.png|width=883,height=48!

Maybe we are missing something? Or something changed? Thank you!

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Hey Valentina Virlics , got it. Thanks for explaining that, I had not gone thorugh the new process yet. So I was a little confused. Anyways, thanks also for testing.

So Android data goes to a different table in Big Query. I am talking to the data folks to try and bring all of these datasets together, but that is not done. Until then, you need to query moz-fx-data-shared-prod.org_mozilla_firefox_vpn_live.main_v1 when verifying Android data.

Weirdly though I was not able to find that ping-id you showed in your print 🤔 Would you mind checking this once more? I can see pings sent from my own Android build in there, that is why I ask.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

We’ve also checked Windows pings, same results (windows ping - d19a79d7-02e4-4c1a-b436-f15749b2dfd9).

Here are the logs from both Android and Windows. (attaching)

[^mozillavpn-2023-1-13_1-win.txt]

[^MozillaVPN_Logs_2023-57-13-11-57-55.txt]

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Hey Beatriz Rizental Machado , thank you for all the details!

I’ve changed the ping. I’ve changed the table and the event is there 😄 like magic!

!image-20230116-153332.png|width=1006,height=672!

Used ping “c18ea082-15d8-4b0c-8b5d-1843f961eb0b“.

So, seems ok on Android.

On Windows, I’ve used the ping ID - bf2fdaf2-ffaa-4c81-9911-9a3c9e525ee8 - and the initial db, but still no events - should I wait?

!image-20230116-153817.png|width=902,height=707!

Thank you!

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

I’ve searched that ping today, and the event is listed. So, you were right, just need to wait a bit.

Based on below comments and verification we can mark this as done.

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Beatriz Rizental Machado this one should be 2.14?
